Mathlib/CategoryTheory/Limits/Shapes/Reflexive.lean

```diff
@@ -148,6 +171,372 @@ instance (priority := 100) hasCoreflexiveEqualizers_of_hasEqualizers [HasEqualiz
+    StructuredArrow.mk (Y := WalkingParallelPair.zero) (id WalkingReflexivePair.one)
+  set e₁ : (StructuredArrow WalkingReflexivePair.zero inclusionWalkingReflexivePair) :=
+    StructuredArrow.mk (Y := WalkingParallelPair.one) (id WalkingReflexivePair.zero)
+  intro x
+  have h : Inhabited (StructuredArrow x inclusionWalkingReflexivePair) := by
+    · constructor
+      cases x with
+      | one => exact e₀
+      | zero => exact e₁
+  cases x with
+    | zero => apply IsConnected.of_induct (j₀ := e₁)
+              intro p h₁ h₂ t
+              rcases t with ⟨l, y, f⟩
+              cases y <;> cases f
+              · set r : StructuredArrow WalkingReflexivePair.zero inclusionWalkingReflexivePair :=
+                  StructuredArrow.mk (Y := WalkingParallelPair.zero) reflexion
+                change r ∈ p
+                suffices f : r ⟶  e₁ by exact (h₂ f).mpr h₁
+                exact StructuredArrow.homMk (WalkingParallelPairHom.left)
+              · exact h₁
+    | one => apply IsConnected.of_induct (j₀ := e₀)
+             intro p h₁ h₂ t
+             rcases t with ⟨l, y, f⟩
+             set rₗ : StructuredArrow WalkingReflexivePair.one inclusionWalkingReflexivePair :=
+                StructuredArrow.mk (Y := WalkingParallelPair.one) left
+             set rᵣ : StructuredArrow WalkingReflexivePair.one inclusionWalkingReflexivePair :=
+                StructuredArrow.mk (Y := WalkingParallelPair.one) right
+             have hrₗ : rₗ ∈ p := by
+              · suffices f : e₀ ⟶  rₗ by exact (h₂ f).mp h₁
+                exact StructuredArrow.homMk WalkingParallelPairHom.left
+             have hrᵣ : rᵣ ∈ p := by
+              · suffices f : e₀ ⟶  rᵣ by exact (h₂ f).mp h₁
+                exact StructuredArrow.homMk WalkingParallelPairHom.right
+             cases y <;> cases f
+             rotate_right 3
+             · exact h₁
+             · exact hrₗ
+             · exact hrᵣ
+             · set v : StructuredArrow WalkingReflexivePair.one inclusionWalkingReflexivePair :=
+                StructuredArrow.mk (Y := WalkingParallelPair.zero) left_reflexion
+               suffices f : v ⟶  rₗ by exact (h₂ f).mpr hrₗ
+               exact StructuredArrow.homMk (WalkingParallelPairHom.left)
+             · set v : StructuredArrow WalkingReflexivePair.one inclusionWalkingReflexivePair :=
+                StructuredArrow.mk (Y := WalkingParallelPair.zero) right_reflexion
+               suffices f : v ⟶  rᵣ by exact (h₂ f).mpr hrᵣ
+               exact StructuredArrow.homMk (WalkingParallelPairHom.right)
+
+end WalkingParallelPair
+
+end Limits
+
+namespace Limits
+
+open WalkingReflexivePair
+
+open WalkingReflexivePairHom
+
+variable {C : Type u} [Category.{v} C]
+
+variable {A B : C}
+
+/-- Bundle the data of a parallel pair along with a common section as a functor out of the walking
+reflexive pair -/
+def reflexivePair (f g : A ⟶ B) (s : B ⟶ A) (sl : s ≫ f = 𝟙 B) (sr : s ≫ g = 𝟙 B) :
```

Comment:
With this, Lean will try to fill in the proofs automatically: 
```suggestion
def reflexivePair (f g : A ⟶ B) (s : B ⟶ A) (sl : s ≫ f = 𝟙 B := by aesop_cat) (sr : s ≫ g = 𝟙 B := by aesop_cat) :
```
Then, you should also add lemmas like
```lean
@[simp] lemma reflexivePair_obj_zero : (reflexivePair f g s sl sr).obj zero = B := rfl
```
and similarly `obj_one`, `map_left`, `map_right` and `map_reflexion`.